sql/catalyst/src/main/java/org/apache/spark/sql/connector/util/V2ExpressionSQLBuilder.java

```diff
@@ -48,6 +48,26 @@
  */
 public class V2ExpressionSQLBuilder {
 
+  protected String escapeSpecialCharsForLikePattern(String str) {
+    StringBuilder builder = new StringBuilder();
+
+    for (char c : str.toCharArray()) {
+      switch (c) {
+        case '_':
+          builder.append("\\_");
+          break;
+        case '%':
+          builder.append("\\%");
+          break;
+        // TODO Support for escaping more special characters
```

Comment:
It's not a TODO. We should document `escapeSpecialCharsForLikePattern` and ask dialects to override it if they have more special chars other than `_` and `%`.